editor.planx.uk/src/pages/FlowEditor/components/PreviewBrowser.tsx

```diff
@@ -68,28 +69,71 @@ const DebugConsole = () => {
   );
 };
 
-function PublishChangeItem(props: any) {
-  const { node } = props;
-  let text, data;
+interface AlteredNodesSummary {
+  title: string;
+  portals: string[];
+  updated: number;
+  deleted: number;
+}
+
+function AlteredNodesSummaryContent(props: any) {
+  const { alteredNodes, url } = props;
+  const changeSummary: AlteredNodesSummary = {
+    title: "",
+    portals: [],
+    updated: 0,
+    deleted: 0,
+  };
 
-  if (node.id === "_root") {
-    text = "Changed _root service by adding, deleting or re-ordering nodes";
-  } else if (node.id === "0") {
-    text = `The entire _root service will be published for the first time`;
-  } else if (node.id && Object.keys(node).length === 1) {
-    text = `Deleted node ${node.id}`;
-  } else if (node.type && node.data) {
-    text = `Added/edited ${TYPES[node.type]}`;
-    data = JSON.stringify(node.data, null, 2);
-  } else {
-    text = `Added/edited ${TYPES[node.type]}`;
-  }
+  alteredNodes.map((node: any) => {
+    if (node.id === "0") {
+      changeSummary["title"] =
+        "You are publishing the main service for the first time.";
+    } else if (node.id && Object.keys(node).length === 1) {
+      changeSummary["deleted"] += 1;
+    } else if (node.type === TYPES.InternalPortal) {
+      if (node.data?.text?.includes("/")) {
+        changeSummary["portals"].push(node.data?.text);
+      }
+    } else if (node.type) {
+      changeSummary["updated"] += 1;
+    }
+
+    return changeSummary;
+  });
 
   return (
-    <>
-      <Typography variant="body2">{text}</Typography>
-      {data && <pre style={{ fontSize: ".8em" }}>{data}</pre>}
-    </>
+    <Box pb={2}>
+      {changeSummary["title"] && (
+        <Typography variant="body2">{changeSummary["title"]}</Typography>
+      )}
+      {(changeSummary["updated"] > 0 || changeSummary["deleted"] > 0) && (
+        <ul>
+          <li key={"updated"}>
+            <Typography variant="body2">{`${changeSummary["updated"]} nodes have been updated or added`}</Typography>
+          </li>
+          <li key={"deleted"}>
+            <Typography variant="body2">{`${changeSummary["deleted"]} nodes have been deleted`}</Typography>
+          </li>
+        </ul>
+      )}
+      {changeSummary["portals"].length > 0 && (
+        <>
+          <Typography variant="body2">{`This includes recently published changes in the following nested services:`}</Typography>
+          <ul>
+            {changeSummary["portals"].map((portal, i) => (
+              <li key={i}>
+                <Typography variant="body2">{portal}</Typography>
```

Comment:
Could making these links be a quick UI win?

Reply:
Yep - but I'm only going to link them if `canUserEditTeam()` is true I think - as it's quite misleading to be linked to a portal you don't actually have permission to publish yourself? 

Same for change badges on portals - I don't think it feels as valuable to show that there's changes waiting to be published if you don't actually have permission to do the publishing? Versus only showing for the flow you've navigated direclty into feels a bit more straightforward at least in this first iteration?

Reply:
Yep, let's keep it simple at first and see what's requested next / proves helpful.